src/ngMessages/messages.js

```diff
@@ -496,31 +496,16 @@ angular.module('ngMessages', [])
     * @param {string} ngMessagesInclude|src a string value corresponding to the remote template.
     */
    .directive('ngMessagesInclude',
-     ['$templateRequest', '$document', '$compile', function($templateRequest, $document, $compile) {
+     ['$templateRequest', '$compile', function($templateRequest, $compile) {
 
      return {
        restrict: 'AE',
-       require: '^^ngMessages',
-       compile: function(element, attrs) {
-         var comment = jqLite($document[0].createComment(' ngMessagesInclude: '));
-         element.after(comment);
```

Comment:
directives are evaluated in the deterministic order, but they'll stop when they reach the terminal one. similarly, child nodes of an element with a terminal directive won't be compiled (but there are no children, except the ones you load from the server and compile on your own.)